### PR TITLE
refactor: Improve uniformity of VisualizerInterface and P5Visualizer

### PR DESCRIPTION
**Pull request by gwhitney**
Three related interface changes for the Visualizer hierarchy, as
   discussed in #293. These are all designed to make some of the
   details of defining a Visualizer less confusing and more concise.

   (A) In P5Visualizer, draw() should be abstract and hence required
       on runnable derived Visualizers, since any (additional) boilerplate
       needed by all P5Visualizers would go in the function that wraps
       this method in the default inhabit() method. Thus implementation
       draw() functions no longer need to call super.draw().
   (B) Since setup() is the default p5 initialization method and can only
       be called once per sketch anyway, clarify that generally initialization
       should be carried out in setup() rather than inhabit(), in derived
       Visualizers.
   (C) Since `name` is used as an instance property in SequenceInterface
       indicating the details of the particular interface, re-designate
       that information as the return value of a `visualization()` method
       of an entity implementing the VisualizationInterface, and suggest
       that it should only depend on the class of the entity, not the
       instance. Implement this in the P5Visualizer hierarchy with
       the visualizationName static property, and modify the
       VisualizerExportModule constructor to look for it there, so that
       the visualizationName will not have to be stated twice in every
       module implementing a Visualizer. At such time as there are other
       base classes for concrete Visualizers, these conventions should
       probably be moved one level up to a common base class of all
       visualizers; there is nothing specifically related to p5 about this
       naming scheme.

By submitting this PR, I am indicating to the Numberscope maintainers that I have read and understood the [contributing guidelines](https://numberscope.colorado.edu/doc/CONTRIBUTING/) and that this PR follows those guidelines to the best of my knowledge. I have also read the [pull request checklist](https://numberscope.colorado.edu/doc/doc/pull-request-checklist/) and followed the instructions therein.

### CODE REVIEW

**Comment by gwhitney** [comment]
(A) Oops, I accidentally pushed to origin rather than my fork when creating this PR. I will try to be more careful.

(B) There were likely some doc-page changes that should go with this, but my belief is that Aaron will merge or rebase this into #293 where there are overlapping doc changes and straighten it all out. @Vectornaut if that's not right, let me know, and I can add a doc commit to this PR as well.

File: src/visualizers/Grid.ts
Comment (katestange) [inline src/visualizers/Grid.ts:401]: why spacing?

File: src/visualizers/ModFill.ts
Comment (katestange) [inline src/visualizers/ModFill.ts:83]: the other vis's have a description at the export line.  For this one a reasonable desc would be 'A dynamic histogram of modular values.' (If I'm recalling the concept correctly.)

File: src/visualizers/NumberGlyph.ts
Comment (katestange) [inline src/visualizers/NumberGlyph.ts:405]: Again, the others have a desc, maybe something like 'Colorful glyphs representing properties of the first few terms'

File: src/visualizers/Turtle.ts
Comment (katestange) [inline src/visualizers/Turtle.ts:164]: maybe 'Terms of the sequence control a colourful walk on the screen.' or 'Turtle walks according to terms in the sequence, creating a trail.'

**Review by katestange — CHANGES_REQUESTED** [review]
Works in production and dev mode.

File: src/visualizers/Grid.ts
Comment (gwhitney) [reply src/visualizers/Grid.ts]: Actually, this reflects a change in how our automated tools reformat Grid.ts that apparently resulted from the giant dependency update we have been going through. I am not quite sure how it filtered through to be a diff in this PR, but now that I have rebased, it is gone as a difference.

File: src/visualizers/ModFill.ts
Comment (gwhitney) [reply src/visualizers/ModFill.ts:83]: OK. The description is optional, and is unused in the current app, but it could be like a tooltip or something in a revamped UI. The fact there is not one here just reflects that there was not one before for ModFill.  But I totally get how this might be good juncture to fill all the rest in. I will add a commit doing that. Do you also want me to make the description mandatory?

**Comment by gwhitney** [comment]
OK, all visualizers have a description. Should we enforce this going forward, or leave it optional as it has been to date?

**Comment by katestange** [comment]
> OK, all visualizers have a description. Should we enforce this going forward, or leave it optional as it has been to date?

I don't know... Yes, I'm imagining this might be a tooltip text or populate some type of menu or explorer or some type.  I leave it to you to decide if it's useful to make mandatory here.  It could be something that depends on the UI revamp.

File: src/visualizers/ShiftCompare.ts
Comment (katestange) [inline src/visualizers/ShiftCompare.ts:120]: whether or not they have a period should probably be consistent.  right now some do, some don't

**Review by katestange — APPROVED** [review]
tiny comment

**Comment by gwhitney** [comment]
OK, I will remove any periods from descriptions that have them since I think they are mostly not full sentences. And I will make descriptions mandatory, since we now have them all and they seem like they might be handy in a UI revamp, if we ever have a chance to do one ;-)

**Comment by katestange** [comment]
looks good!

**Comment by gwhitney** [comment]
OK, Aaron, ready for you to push the merge button whenever it works for you.

**Comment by Vectornaut** [comment]
> Implement this in the P5Visualizer hierarchy with
the visualizationName static property, and modify the
VisualizerExportModule constructor to look for it there, so that
the visualizationName will not have to be stated twice in every
module implementing a Visualizer.

I like this simplification!

File: src/components/BundleCard.vue
Comment (Vectornaut) [inline src/components/BundleCard.vue:1]: The new messages are much clearer to me.

File: src/components/BundleCard.vue
Comment (Vectornaut) [inline src/components/BundleCard.vue:10]: These new messages are much clearer to me.

**Review by Vectornaut — APPROVED** [review]

**Pull request merged**
